test/test-integration/src/test/java/org/wso2am/micro/gw/tests/common/BaseTestCase.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2am.micro.gw.tests.common;
+
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2am.micro.gw.tests.common.model.API;
+import org.wso2am.micro.gw.tests.common.model.ApplicationDTO;
+import org.wso2am.micro.gw.tests.common.model.SubscribedApiDTO;
+import org.wso2am.micro.gw.tests.context.MgwServerInstance;
+import org.wso2am.micro.gw.tests.context.MicroGWTestException;
+import org.wso2am.micro.gw.tests.util.*;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+
+/**
+ * Base test class for mgw test cases.
+ */
+public class BaseTestCase {
+
+    protected MgwServerInstance microGWServer;
+
+    /**
+     * start the mgw docker environment and mock backend.
+     *
+     * @throws MicroGWTestException
+     * @throws IOException
+     */
+    public void startMGW() throws MicroGWTestException, IOException, InterruptedException {
+        microGWServer = new MgwServerInstance();
+        microGWServer.startMGW();
+    }
+
+    /**
+     * start the mgw docker environment and mock backend.
+     *
+     * @param confPath       external conf.toml file location
+     *
+     * @throws MicroGWTestException
+     * @throws IOException
+     */
+    public void startMGW(String confPath) throws MicroGWTestException, IOException, InterruptedException {
+        microGWServer = new MgwServerInstance(confPath);
+        microGWServer.startMGW();
+    }
+
+    /**
+     * stop the mgw docker environment.
+     */
+    public void stopMGW() {
+        microGWServer.stopMGW();
+
+    }
+
+    public static String getImportAPIServiceURLHttps(String servicePath) throws MalformedURLException {
+        return new URL(new URL("https://localhost:" + TestConstant.ADAPTER_IMPORT_API_PORT), servicePath)
+                .toString();
+    }
+
+    public static String getServiceURLHttps(String servicePath) throws MalformedURLException {
+        return new URL(new URL("https://localhost:" + TestConstant.GATEWAY_LISTENER_HTTPS_PORT), servicePath)
+                .toString();
+    }
+
+    public static String getMockServiceURLHttp(String servicePath) throws MalformedURLException {
+        return new URL(new URL("http://localhost:" + TestConstant.MOCK_SERVER_PORT), servicePath).toString();
+    }
+
+    /**
+     * get a jwt token.
+     *
+     * @param api                 api
+     * @param applicationDTO      application dto
+     * @param tier                tier
+     * @param keyType             keytype
+     * @param validityPeriod      validityPeriod
+     *
+     * @throws Exception
```

Comment:
Shall we dd the Javadoc for return type as well?

Reply:
Fixed in 3b6cc6755810d748cd3de4fb25696a5692c08e78